src/Service/SmsService.php

```diff
@@ -0,0 +1,131 @@
+    public function __construct(
+        Environment $twig
+    ) {
+        $this->twig = $twig;
+
+    }//end __construct()
+
+
+    /**
+     * Handles the sending of an email based on an event.
+     *
+     * @param array $data
+     * @param array $configuration
+     *
+     * @throws LoaderError|RuntimeError|SyntaxError|TransportExceptionInterface
+     *
+     * @return array
+     */
+    public function SmsHandler(array $data, array $configuration): array
+    {
+        $this->data          = $data;
+        $this->configuration = $configuration;
+
+        $this->sendSms();
+
+        return $data;
+
+    }//end SmsHandler()
+
+
+    /**
+     * Sends and email using an EmailTemplate with configuration for it. It is possible to use $object data in the email if configured right.
+     *
+     * @throws LoaderError
+     * @throws SyntaxError
+     * @throws TransportExceptionInterface
+     *
+     * @return bool
+     */
+    private function sendSms(): bool
+    {
+        // Ready the email template with configured variables
+        $variables = [];
+
+        $dataDot = new Dot($this->data);
+        foreach ($this->configuration['variables'] as $key => $variable) {
+            // Response is the default used for creating emails after an /api endpoint has been called and returned a response.
+            if ($dataDot->has('response'.$variable) === true) {
+                $variables[$key] = $dataDot->get('response'.$variable);
+                continue;
+            }
+
+            if ($dataDot->has($variable) === true) {
+                $variables[$key] = $dataDot->get($variable);
+                continue;
+            }
+
+            if ((str_contains($variable, '{%') === true && str_contains($variable, '%}') === true)
+                || (str_contains($variable, '{{') === true && str_contains($variable, '}}') === true)
+            ) {
+                $variables[$key] = $this->twig->createTemplate($variable)->render($variables);
+            }
+        }
+
+        // Render the template
+        $html = $this->twig->createTemplate(base64_decode($this->configuration['template']))->render($variables);
+        $text = strip_tags(preg_replace('#<br\s*/?>#i', "\n", $html), '\n');
+
+        // Lets allow the use of values from the object Created/Updated with {attributeName.attributeName} in the these^ strings.
+        $subject  = $this->twig->createTemplate($this->configuration['subject'])->render($variables);
+        $receiver = $this->twig->createTemplate($this->configuration['receiver'])->render($variables);
+        $sender   = $this->twig->createTemplate($this->configuration['sender'])->render($variables);
+
+        // Create mailer with mailgun url
+        $transport = Transport::fromDsn($this->configuration['serviceDNS'].'?from='.$sender);
+        $texter    = new Texter($transport);
+
+        // If we have no sender, set sender to receiver
+        if (!$sender) {
+            $sender = $receiver;
+        }
```

Comment:
Don't we want to throw an error if the sender is not given?

Reply:
Good point, this was a copy paste from the e-mail software, but it's good to just log an error and exit if we do not have a sender.

Reply:
Please also add this log to the emailService as well